### DISCUSSION TITLE
FlowGrid Extension

### DISCUSSION THREAD

**Discussion by sveinhelge** [QuestPDF/QuestPDF#60] [Ideas]
Let's say we have a grid with 3 columns.
I want to fill all columns with block of texts.

My data are:
1. Text
2. Text
    Text
3. Text
4. Text
5. Text
6. Text
7. Text
    Text
    Text
8. Text
9. Text
    Text
10. Text

This is the result I would like to have:
1 Text | 3 Text | 7 Text
2 Text | 4 Text |    Text
   Text | 5 Text |    Text
---Page Break---
8 Text | 10 Text
9 Text
   Text
    
Since the number of elements and length of texts are dynamic I would like the block of texts to float vertically.
So It should fill the first column first as long as it can fit on the same page. If not, it should add the text block in the next column.
When all columns are full, move to the next page.

Then we would of course have scenarios where one text block will not fit in one entire column on one page. We could then spread that text over into the next column on the next page. Then start with the next text block in the next column. In the first versjon it would be ok if the rendering breaks if you add a text block which would not fit on an entire page. Leaving that responsibility to the document creator.

**Comment by MarcinZiabek** [comment]
This looks like a totally new layouting element. As far as I understand, this is equivalent to [the multi-columns layout in CSS](https://www.w3schools.com/css/css3_multiple_columns.asp). 

I must confess, this one is tricky as it introduces a new kind of wrapping. Right now, we support wrapping only between pages. In here, we would need to support wrapping against columns AND pages. 

Without some code changes in most components, this may be difficult to implement. I will put it on the roadmap somewhere 😀 If anybody else needs such functionality, please leave a comment - this will help me prioritize this feature.

**Reply by sveinhelge** [reply]
Yes, it would be very much equivalent to the multi-columns layout with **break-inside: avoid;**  per block of text.
One option could be to build the new component in steps. First version does not support paging. The second version will support it.
I'm happy it's on the roadmap.

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

The functionality you described is now available through the [MultiColumn](https://www.questpdf.com/api-reference/multi-column.html) element, introduced in version 2024.7.0.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
